### PR TITLE
Random Battles: Improve documentation

### PR DESCRIPTION
**Pull request by ACakeWearingAHat**
(and also fix gen 3 STAB enforcement, since i've already finished it)

intent: use more thorough commenting and easier-to-understand names in order to make move enforcement code less cryptic to people who don't know code that well

### CODE REVIEW

File: data/mods/gen3/random-teams.ts [hunk 1]
```diff
@@ -371,43 +371,60 @@ export class RandomGen3Teams extends RandomGen4Teams {
...
+					!hasMove['seismictoss'] && !hasMove['nightshade'] &&
+					species.id !== 'castform' &&
+					// If a Flying-type has Psychic, it doesn't need STAB
+					!(hasMove['psychic'] && hasType['Flying']) &&
+					!(hasType['Ghost'] && species.baseStats.spa > species.baseStats.atk) &&
+					!(
+						// With Calm Mind, Lugia and pure Normal-types are fine without STAB
+						counter.setupType === 'Special' &&
+						species.id === 'lugia' ||
+						(hasType['Normal'] && !species.types[1])
```
Comment (AnnikaCodes) [inline data/mods/gen3/random-teams.ts:394]:
```suggestion
						(hasType['Normal'] && species.types.length < 2)
```

File: data/mods/gen3/random-teams.ts [hunk 2]
```diff
@@ -371,43 +371,60 @@ export class RandomGen3Teams extends RandomGen4Teams {
...
+					) &&
+					!(
+						// With Swords Dance, Dark-types and pure Water-types are fine without STAB
+						counter.setupType === 'Physical' &&
+						((hasType['Water'] && !species.types[1]) || hasType['Dark'])
```
Comment (AnnikaCodes) [inline data/mods/gen3/random-teams.ts:399]:
```suggestion
						((hasType['Water'] && species.types.length < 2) || hasType['Dark'])
```

File: data/mods/gen3/random-teams.ts (hunk 1 above)
Comment (ACakeWearingAHat) [reply data/mods/gen3/random-teams.ts:394]: i was just being consistent with other parts of the file honestly but sure

**Comment by AnnikaCodes** [comment]
Looks good, thanks!

**Pull request merged**